### PR TITLE
Stop steps after they are done

### PR DESCRIPTION
**Pull request by anbraten**
We haven't stopped steps immediateley after they were done. Therefore in case of the docker & kubernetes backends the containers of finished steps existed til the whole workflow was done.

### CODE REVIEW

File: pipeline/backend/types/engine.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type Engine interface {
 	// TailStep the workflow step logs.
 	TailStep(ctx context.Context, step *Step, taskUUID string) (io.ReadCloser, error)
 
+	// StopStep stop the workflow step.
+	StopStep(ctx context.Context, step *Step, taskUUID string) error
```
Comment (6543) [inline pipeline/backend/types/engine.go:48]: we should call it what it is: DestroyStep ?

Comment (6543) [reply pipeline/backend/types/engine.go:48]: or more CleanStep

Comment (anbraten) [reply pipeline/backend/types/engine.go:48]: Wanted to be consistent with `StartStep` which is also a `create` somehow.

**Comment by 6543** [comment]
if a container is already deleted ... does this lead to errors if DestoryWorkflow is exec ?

**Comment by anbraten** [comment]
> DestoryWorkflow

We have `isErrContainerNotFoundOrNotRunning` checks in the `DestroyWorkflow` part, so that should be fine.

File: pipeline/backend/types/engine.go (hunk 1 above)
Comment (anbraten) [reply pipeline/backend/types/engine.go:48]: Renamed it to `DestroyStep` as its also `DestroyWorkflow`

File: pipeline/backend/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -325,6 +325,29 @@ func (e *kube) TailStep(ctx context.Context, step *types.Step, taskUUID string)
...
+
+	if err := e.client.CoreV1().Pods(e.config.Namespace).Delete(ctx, podName, deleteOpts); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	return nil
+}
+
 // Destroy the pipeline environment.
 func (e *kube) DestroyWorkflow(_ context.Context, conf *types.Config, taskUUID string) error {
```
Comment (6543) [inline pipeline/backend/kubernetes/kubernetes.go:352]: could we remove the log in 376  as it is default now

Comment (anbraten) [reply pipeline/backend/kubernetes/kubernetes.go:352]: done

**Review by 6543 — APPROVED** [review]

**Comment by strk** [comment]
Will this be backported to 1.0 branch too ?

**Pull request merged**